pkg/apis/sessions/session_state.go

```diff
@@ -102,32 +102,39 @@ func DecodeSessionState(v string, c *encryption.Cipher) (*SessionState, error) {
 			PreferredUsername: ss.PreferredUsername,
 		}
 	} else {
-		// Backward compatibility with using unencrypted Email
-		if ss.Email != "" {
-			decryptedEmail, errEmail := c.Decrypt(ss.Email)
-			if errEmail == nil {
-				ss.Email = decryptedEmail
-			}
-		}
-		// Backward compatibility with using unencrypted User
-		if ss.User != "" {
-			decryptedUser, errUser := c.Decrypt(ss.User)
-			if errUser == nil {
-				ss.User = decryptedUser
-			}
-		}
+		// Backward compatibility with using unencrypted Email or User
+		// Decryption errors will leave original string
+		_ = into(&ss.Email, c.Decrypt)
+		_ = into(&ss.User, c.Decrypt)
```

Comment:
I think what we'd actually like to do is drop the backward compatibility, we discussed this in #601, basically, if we have an issue decrypting, or if the decrypted result is not valid UTF 8, then we believe that the session was previously unencrypted and decided that we should return an error, to force users to re-authenticate in this case.

So what we could do here, is just move Email and User into the range statement below I think, would have the same sort of effect right? 

CC @steakunderscore can you also confirm my logic and this is what we had agreed upon

Reply:
Now that #601 has merged I'll rebase and incorporate those changes here 👍

Reply:
aaaaaand rebased.  I think I captured the spirit of #601 with how we check the errors via the `into` methods